lyaemu/deltaF_pdf.py

```diff
@@ -0,0 +1,226 @@
+"""Modules to generate the flux power spectrum from a simulation box."""
+from __future__ import print_function
+import argparse
+import os.path
+import scipy.interpolate
+import numpy as np
+from fake_spectra import spectra
+from fake_spectra import abstractsnapshot as absn
+import spectra_mocking as sm
+
+def rebin_power_to_kms(kfkms, kfmpc, flux_powers, zbins, omega_m, omega_l = None):
+    """Rebins a power spectrum to constant km/s bins.
+    Bins larger than the box are discarded. The return type is thus a list,
+    with each redshift bin having potentially different lengths."""
+    if omega_l is None:
+        omega_l = 1 - omega_m
+    nz = np.size(zbins)
+    nk = np.size(kfmpc)
+    assert np.size(flux_powers) == nz * nk
+    velfac = lambda zz: 1./(1+zz) * 100.0* np.sqrt(omega_m * (1 + zz)**3 + omega_l)
+    rebinned=[scipy.interpolate.interpolate.interp1d(kfmpc,flux_powers[ii*nk:(ii+1)*nk]) for ii in range(nz)]
+    okmsbins = [kfkms[np.where(kfkms >= np.min(kfmpc)/velfac(zz))] for zz in zbins]
+    flux_rebinned = [rebinned[ii](okmsbins[ii]*velfac(zz)) for ii, zz in enumerate(zbins)]
+    return okmsbins, flux_rebinned
+
+class FluxPdf(object):
```

Comment:
Why don't you subclass the FluxPower object and re-use the common functions? As in:

class FluxPdf(FluxPower):

and then only get_deltaF_pdf needs to change.